src/goto-symex/symex_complexity.cpp

```diff
@@ -0,0 +1,229 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: John Dumbell
+
+\*******************************************************************/
+
+#include "symex_complexity.h"
+#include "goto_symex_state.h"
+#include <cmath>
+
+symex_complexityt::symex_complexityt(
+  message_handlert &logger,
+  const optionst &options)
+  : log(logger)
+{
+  std::size_t limit = options.get_signed_int_option("symex-complexity-limit");
+  if((complexity_active = limit > 0))
+  {
+    // This gives a curve that allows low limits to be rightly restrictive,
+    // while larger numbers are very large to represent what you're asking.
+    max_complexity = static_cast<std::size_t>((limit * limit) * 25);
+
+    const std::size_t failed_children_limit =
+      options.get_signed_int_option("symex-complexity-failed-children-limit");
+    const std::size_t unwind = options.get_signed_int_option("unwind");
+
+    // In order of priority:
+    //  * explicit limit
+    //  * inferred limit from unwind
+    //  * best limit we can apply with very little information.
+    if(failed_children_limit > 0)
+      max_loops_complexity = failed_children_limit;
+    else if(unwind > 0)
+      max_loops_complexity = std::max(static_cast<int>(floor(unwind / 3)), 1);
```

Comment:
justify the /3 somehow?

Reply:
2/7, 1/3, 3/10 are all decent numbers for how many branches you'd allow to fail when using this system. Nothing more than that really - if you want more refined control I'd expect you to pass an exact number.